params/hooks_libevm.go

```diff
@@ -79,15 +82,18 @@ var PrecompiledContractsGranite = map[common.Address]vm.PrecompiledContract{
 	nativeasset.GenesisContractAddr:    makePrecompile(&nativeasset.DeprecatedContract{}),
 	nativeasset.NativeAssetBalanceAddr: makePrecompile(&nativeasset.DeprecatedContract{}),
 	nativeasset.NativeAssetCallAddr:    makePrecompile(&nativeasset.DeprecatedContract{}),
+	P256VerifyAddress:                  vm.PrecompiledContractsP256Verify[P256VerifyAddress],
 }
 
-func init() {
-	for addr, precompile := range vm.PrecompiledContractsP256Verify {
-		PrecompiledContractsGranite[addr] = precompile
-	}
+// ActivePrecompiles implements the RulesHooks interface
+func (r RulesExtra) ActivePrecompiles(existing []common.Address) []common.Address {
+	addresses, _, _ := r.ActivePrecompilesWithContract(existing, nil)
```

Comment:
The fact that you're dropping the last 2 return arguments and you're passing a `nil` argument into the function are suggestive of an inappropriate abstraction (a.k.a. a [code smell](https://en.wikipedia.org/wiki/Code_smell)), which forces the caller to "massage" the code into doing what's necessary instead of it just flowing.

Have a look at https://github.com/ava-labs/coreth/commit/79032da35 for an alternative abstraction (the `currentPrecompiles()` method). It's based off the commit that I originally reviewed, not this current one, so you can more clearly see what I've done.